pkg/master/master.go

```diff
@@ -188,7 +158,10 @@ type thirdPartyEntry struct {
 }
 
 type RESTOptionsGetter func(resource unversioned.GroupResource) generic.RESTOptions
-type RESTStorageProvider func(apiResourceConfigSource genericapiserver.APIResourceConfigSource, restOptionsGetter RESTOptionsGetter) (genericapiserver.APIGroupInfo, bool)
+
+type RESTStorageProvider interface {
+	NewRESTStorage(apiResourceConfigSource genericapiserver.APIResourceConfigSource, restOptionsGetter RESTOptionsGetter) (genericapiserver.APIGroupInfo, bool)
```

Comment:
> Why the bool in the result? If I am not mistaken, that value is true for all implementations.

These implementations don't depend on any conditional information, but as you have endpoints based on things like authenticators and authorizers which are optional, `false` because a valid response.  "couldn't built it because I'm missing the inputs".  I could maybe see an argument for an `error` instead, but some "I failed" marker must exist.